lib/addons/amp/index.ts

```diff
@@ -0,0 +1,131 @@
+import { KubernetesManifest } from "aws-cdk-lib/aws-eks";
+import * as aps from 'aws-cdk-lib/aws-aps';
+import { CfnWorkspaceProps } from 'aws-cdk-lib/aws-aps';
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { dependable, loadYaml, readYamlDocument } from "../../utils";
+import { AdotCollectorAddOn } from "../adot";
+import { Construct } from 'constructs';
+
+/**
+ * Configuration options for add-on.
+ */
+ export interface AmpAddOnProps {
+    /**
+     * Name that will be used by the Addon to create the Workspace
+     * @default blueprints-amp-workspace
+     */
+    workspaceName?: string;
+    /** 
+     * Remote Write URL of the AMP Workspace to be used for setting up remote write.
+     */
+     prometheusRemoteWriteURL?: string;
+    /**
+     * Tag Key to passed while creating AMP workspace
+     * @default Project
+     */
+     workspaceTagKey?: string;
+    /**
+     * Tag Value to passed while creating AMP workspace
+     * @default blueprints-amp-workspace
+     */
+     workspaceTagValue?: string;
+    /**
+     * Modes supported : `deployment`, `daemonset`, `statefulSet`, and `sidecar`
+     * @default deployment
+     */
+     deploymentMode?: DeploymentMode;
+}
+
+export const enum DeploymentMode {
+    DEPLOYMENT = 'deployment',
+    DAEMONSET = 'daemonset',
+    STATEFULSET = 'statefulset',
+    SIDECAR = 'sidecar'
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps = {
+    workspaceName: 'blueprints-amp-workspace',
+    workspaceTagKey: 'Project',
+    workspaceTagValue: 'blueprints-amp-workspace',
+    deploymentMode: DeploymentMode.DEPLOYMENT
+};
+
+/**
+ * Implementation of AMP add-on for EKS Blueprints. Installs ADOT Collector.
+ */
+export class AmpAddOn implements ClusterAddOn {
+
+    readonly ampAddOnProps: AmpAddOnProps;
+
+    constructor(props?: AmpAddOnProps) {
+        this.ampAddOnProps = { ...defaultProps, ...props };
+    }
+
+    @dependable(AdotCollectorAddOn.name)
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let finalRemoteWriteURLEndpoint: string;
+        let finalDeploymentMode: string;
+        let doc: string;
+        
+        let cfnWorkspaceProps : CfnWorkspaceProps  = {
+            alias: defaultProps.workspaceName,  
+            tags: [{
+            key: defaultProps.workspaceTagKey,
+            value: defaultProps.workspaceTagValue
+            }],
+        }
```

Comment:
`make lint` will produce warnings. Please fix this:
```
lib/addons/amp/index.ts#L80

Missing semicolon
```

Reply:
All lint issues are fixed, thankyou for pointing out.